packages/side-nav/src/SideNavGroup.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { css } from '@leafygreen-ui/emotion';
+import { uiColors } from '@leafygreen-ui/palette';
+import { ulStyleOverrides, LEFT_RIGHT_OFFSET } from './styles';
+
+const sideNavLabelStyle = css`
+  color: ${uiColors.green.base};
+  text-transform: uppercase;
+  font-size: 12px;
+  margin-top: 0;
+  margin-bottom: 0;
+  padding: 5px ${LEFT_RIGHT_OFFSET}px 5px ${LEFT_RIGHT_OFFSET}px;
+`;
+
+interface SideNavGroupProps {
+  /**
+   * Class name that will be applied to the root-level element.
+   */
+  className?: string;
+
+  /**
+   * Content that will be rendered as the component's header. If a string is provided,
+   * it will be rendered with default styling as a header tag.
+   */
+  header?: React.ReactNode;
+
+  /**
+   * Content that will be rendered inside the root-level element.
+   */
+  children?: React.ReactNode;
+}
+
+/**
+ * # SideNavGroup
+ *
+ * ```
+<SideNavGroup headerText="Section Header">
+  <SideNavItem href="/">
+    Back to Home
+  </SideNavItem>
+</SideNavGroup>
+ * ```
+ *
+ * @param props.className Class name that will be applied to the root-level element.
+ * @param props.header Content that will be rendered as the component's header
+ *   If a string is provided, it will be rendered with default styling as a header tag.
+ * @param props.children Class name that will be applied to the component's header.
+ */
+function SideNavGroup({ header, children, ...rest }: SideNavGroupProps) {
+  return (
+    <li {...rest}>
+      {typeof header === 'string' ? (
+        <h4 className={sideNavLabelStyle}>{header}</h4>
+      ) : (
+        header
```

Comment:
Hmm... I have mixed feelings about not rendering the H4 when the prop isn't a string. Reason being is that if anyone needed to include an icon or span in the header, they'd have to duplicate the header styles entirely...

Reply:
Definitely, good plan. Theoretically if someone wants a custom header without those styles, they should just pass it in as part of `children`.